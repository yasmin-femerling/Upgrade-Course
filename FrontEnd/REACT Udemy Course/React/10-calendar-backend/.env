PORT=3001
DB_CNN=mongodb+srv://mern_user:K6Z2h49lxF50QJsf@cluster0.dtvdp.mongodb.net/mern_calendar
SECRET_JWT_SEED=Esto-Es-Una_Palabra_Secreta123456