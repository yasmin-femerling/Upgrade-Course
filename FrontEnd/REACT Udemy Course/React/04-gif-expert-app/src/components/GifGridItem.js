import PropTypes from 'prop-types';
import React from 'react';

export const GifGridItem = ({ title, url }) => {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={ url } alt={ title } />
            <p> { title } </p>
        </div>
    )
}

/* tarea prueba
mostrar componente correctamente
    * shallow
    *wrapper
    * wrapper .tomatchsnapshot()
*/


GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};