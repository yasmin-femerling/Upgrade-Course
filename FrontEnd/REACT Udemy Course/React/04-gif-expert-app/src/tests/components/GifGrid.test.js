import { GifGrid } from "../../components/GifGrid";
import React from 'react';
import {shallow} from 'enzyme';
import{useFetchGifs} from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs.js');
import '@testing-library/jest-dom';

describe('Pruebas en GifGrid', () => {

  test('debe mostrarse correctamente', () => {
    useFetchGifs.mockReturnValue({
        data:[],
        loading:true,
    });
    const wrapper = shallow(<GifGrid/>);
    expect(wrapper).toMatchSnapshot();
});

test('debe mostrar items cuando se cargan imagenes usando useFetchGifs', () => {
    const gifs = [{
        id:'ABC',
        url:'https://lalala.com'
    }]
    useFetchGifs.mockReturnValue({
        data:gifs,
        loading:false,
    });
    const wrapper = shallow(<GifGrid/>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
});

test('debe', () => {
  ;
});

});
