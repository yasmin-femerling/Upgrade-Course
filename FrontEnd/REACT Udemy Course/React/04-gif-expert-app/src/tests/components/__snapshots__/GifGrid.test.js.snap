// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid debe mostrar items cuando se cargan imagenes usando useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      url="https://lalala.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en GifGrid debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
